library/src/blas2/rocblas_tbsv_strided_batched.cpp

```diff
@@ -0,0 +1,254 @@
+                      uplo,
+                      transA,
+                      diag,
+                      n,
+                      k,
+                      A,
+                      lda,
+                      stride_A,
+                      x,
+                      incx,
+                      stride_x,
+                      batch_count);
+
+        if(layer_mode & (rocblas_layer_mode_log_bench | rocblas_layer_mode_log_profile))
+        {
+            auto uplo_letter   = rocblas_fill_letter(uplo);
+            auto transA_letter = rocblas_transpose_letter(transA);
+            auto diag_letter   = rocblas_diag_letter(diag);
+
+            if(layer_mode & rocblas_layer_mode_log_bench)
+            {
+                if(handle->pointer_mode == rocblas_pointer_mode_host)
+                    log_bench(handle,
+                              "./rocblas-bench -f tbsv_strided_batched -r",
+                              rocblas_precision_string<T>,
+                              "--uplo",
+                              uplo_letter,
+                              "--transposeA",
+                              transA_letter,
+                              "--diag",
+                              diag_letter,
+                              "-n",
+                              n,
+                              "-k",
+                              k,
+                              "--lda",
+                              lda,
+                              "--stride_a",
+                              stride_A,
+                              "--incx",
+                              incx,
+                              "--stride_x",
+                              stride_x,
+                              "--batch_count",
+                              batch_count);
+            }
+
+            if(layer_mode & rocblas_layer_mode_log_profile)
+                log_profile(handle,
+                            rocblas_tbsv_name<T>,
+                            "uplo",
+                            uplo_letter,
+                            "transA",
+                            transA_letter,
+                            "diag",
+                            diag_letter,
+                            "N",
+                            n,
+                            "K",
+                            k,
+                            "lda",
+                            lda,
+                            "stride_a",
+                            stride_A,
+                            "incx",
+                            incx,
+                            "stride_x",
+                            stride_x,
+                            "batch_count",
+                            batch_count);
+        }
+
+        if(uplo != rocblas_fill_lower && uplo != rocblas_fill_upper)
+            return rocblas_status_invalid_value;
+        if(batch_count < 0 || ((n < 0 || k < 0 || lda < k + 1 || !incx) && batch_count > 0))
+            return rocblas_status_invalid_size;
```

Comment:
Not sure why batch_count=0 can overrule invalid args but n=0 does not.   If this was specified in our design doc, I may have forgotten.   If we choose or have chosen to go this way the !batch_count return status_sucess could just be moved ahead of invalid_size checks for clarity.

Reply:
Don't see it anywhere in the design doc or anything.
I remember having the discussion awhile back, the thinking for this was that if batch_count == 0, it shouldn't matter what any of the other parameters are as there's no batches. This is different from n == 0 since there _is_ a valid batch, just that one of the parameters makes it so no calculations are necessary.
This currently matches trsv, I'll leave it for now. Do you know if there's a ticket for the "argument error-checking" task we've discussed in the past? I can make one or add this discussion to it.

Reply:
Lee may have an open ticket but there is enough work to make a new one I think.  I have been coding up batch_count == 0 as an equivalent to other quick returns.  This happens after invalid argument checks, so there is no difference to n==0, simpler in my opinion there is not another rule to consider.   magma dsyr2k_batched implementation does it this way,  should compare against cublas.